src/dac.rs

```diff
@@ -0,0 +1,60 @@
+use riot_sys::dac_t;
+
+#[derive(Debug)]
+pub struct DACLine {
+    line: dac_t,
+}
+
+#[derive(Debug)]
+pub enum DACError {
+    /// The given dac_t line did not exist
+    NoLine,
+    /// An unknown error did occur
+    Unknown,
+}
+
+impl DACLine {
+    /// Creates and intializes a new [`DACLine`] from the given
+    /// unitialized [`dac_t`].
+    ///
+    /// Returns an Error if the given line does not exist
+    /// on the board.
+    pub fn new(line: dac_t) -> Result<Self, DACError> {
+        let res = unsafe { riot_sys::dac_init(line) } as i32;
+
+        const DAC_OK: i32 = riot_sys::DAC_OK as i32;
+        const DAC_NOLINE: i32 = riot_sys::DAC_NOLINE as i32;
+
+        match res {
+            DAC_OK => Ok(DACLine { line }),
+            DAC_NOLINE => Err(DACError::NoLine),
+            _ => Err(DACError::Unknown),
+        }
+    }
+
+    /// Builds a [`DACLine`] from an already initialized [`dac_t`].
+    ///
+    /// Providing a not initialized [`dac_t`] results in undefined behavior.
+    pub unsafe fn new_without_init(line: dac_t) -> Self {
+        DACLine { line }
+    }
+
+    /// Writes the given value to this [`DACLine`]
+    ///
+    /// The `value` is internally scaled to the underlying
+    /// dac device so that the maximum voltage output
+    /// is always equivalent to [`u16::MAX`]
+    pub fn set(&self, value: u16) {
```

Comment:
I'm conflicted about this and the power methods not taking a &mut.

Yes, RIOT provides the abstractions this way, but at the same time, picking the more Rust-idiomatic choice would help keep the Rust interface idiomatic if RIOT should find later that maybe working off the same initialized dac_t simultaneously in different threads is *not* such a good idea (but it just wasn't documented).

What are your thoughts on this?

Reply:
That's a good idea, just did not thought about it. I can't think of any justification for those _not_ to be `mut` in the Rust interface. If you really need to access it from different threads one should use safe Rust ways to do that. 
Ill change those to `mut` references.

Reply:
Done